frontend/src/scenes/feature-flags/FeatureFlags.tsx

```diff
@@ -185,81 +186,111 @@ export function OverViewTab({
                                 >
                                     Copy feature flag key
                                 </LemonButton>
-                                <LemonButton
-                                    data-attr={`feature-flag-${featureFlag.key}-switch`}
-                                    onClick={() => {
-                                        const newValue = !featureFlag.active
-                                        LemonDialog.open({
-                                            title: `${newValue === true ? 'Enable' : 'Disable'} this flag?`,
-                                            description: `This flag will be immediately ${
-                                                newValue === true ? 'rolled out to' : 'rolled back from'
-                                            } the users matching the release conditions.`,
-                                            primaryButton: {
-                                                children: 'Confirm',
-                                                type: 'primary',
-                                                onClick: () => {
-                                                    featureFlag.id
-                                                        ? updateFeatureFlag({
-                                                              id: featureFlag.id,
-                                                              payload: { active: newValue },
-                                                          })
-                                                        : null
-                                                },
-                                                size: 'small',
-                                            },
-                                            secondaryButton: {
-                                                children: 'Cancel',
-                                                type: 'tertiary',
-                                                size: 'small',
-                                            },
-                                        })
-                                    }}
-                                    id={`feature-flag-${featureFlag.id}-switch`}
-                                    disabled={!featureFlag.can_edit}
-                                    fullWidth
+
+                                <AccessControlAction
+                                    userAccessLevel={featureFlag.user_access_level}
+                                    requiredLevels={['admin', 'editor']}
+                                    resourceType="feature flag"
                                 >
-                                    {featureFlag.active ? 'Disable' : 'Enable'} feature flag
-                                </LemonButton>
+                                    {({ disabledReason: accessControlDisabledReason }) => (
+                                        <LemonButton
```

Comment:
One thought - this seems to be generally applied to buttons across the board. Maybe having either a prop for LemonButton or a derived "AccessControlledLemonButton" or something wouldbe nicer long term

Reply:
I was trying to make something reusable for anything because it's also used on switches and such but I could make something specifically for buttons to make it cleaner 👍